contracts/contracts/token/OUSD.sol

```diff
@@ -214,17 +285,29 @@ contract OUSD is Initializable, InitializableToken, Governable {
     function _mint(address _account, uint256 _amount) internal {
         require(_account != address(0), "Mint to the zero address");
 
-        _totalSupply = _totalSupply.add(_amount);
+        bool isNonRebasingAccount = _isNonRebasingAddress(_account);
 
-        uint256 creditAmount = _amount.mulTruncate(creditsPerToken);
+        // If the account is non rebasing and doesn't have a set creditsPerToken
+        // then set it i.e. this is a mint from a fresh contract
+        if (isNonRebasingAccount && nonRebasingCreditsPerToken[_account] == 0) {
+            nonRebasingCreditsPerToken[_account] = creditsPerToken;
+        }
+
+        uint256 creditAmount = _amount.mulTruncate(_creditsPerToken(_account));
         _creditBalances[_account] = _creditBalances[_account].add(creditAmount);
         totalCredits = totalCredits.add(creditAmount);
+        _totalSupply = _totalSupply.add(_amount);
+
+        if (isNonRebasingAccount) {
+            nonRebasingCredits = nonRebasingCredits.add(creditAmount);
```

Comment:
Noob question. Why do we use .add() here while on line 171 for example we use += ?

Reply:
I should be using .add/.sub etc everywhere. That uses the SafeMath library which will revert on overflows, rather than the unchecked + version.